pyanaconda/modules/common/structures/partitioning.py

```diff
@@ -18,10 +18,232 @@
 # Red Hat, Inc.
 #
 
-from pyanaconda.dbus.structure import DBusData
+from pyanaconda.core.constants import DEFAULT_AUTOPART_TYPE
+from pyanaconda.dbus.structure import DBusData, generate_string_from_data
 from pyanaconda.dbus.typing import *  # pylint: disable=wildcard-import
 
-__all__ = ["MountPointRequest"]
+__all__ = ["PartitioningRequest", "MountPointRequest"]
+
+
+class PartitioningRequest(DBusData):
+    """Partitioning request data."""
+
+    def __init__(self):
+        self._partitioning_scheme = DEFAULT_AUTOPART_TYPE
+        self._file_system_type = ""
+        self._excluded_mount_points = []
+
+        self._encrypted = False
+        self._passphrase = ""
+        self._cipher = ""
+        self._luks_version = ""
+
+        self._pbkdf = ""
+        self._pbkdf_memory = 0
+        self._pbkdf_time = 0
+        self._pbkdf_iterations = 0
+
+        self._escrow_certificate = ""
+        self._backup_passphrase_enabled = False
+
+    @property
+    def partitioning_scheme(self) -> Int:
+        """The partitioning scheme.
+
+        Allowed values:
+            0  Create regular partitions.
+            1  Use the btrfs scheme.
+            2  Use the lvm scheme.
+            3  Use the lvm thinp scheme.
+
+        :return: an number of the partitioning scheme
+        """
+        return self._partitioning_scheme
+
+    @partitioning_scheme.setter
+    def partitioning_scheme(self, scheme: Int):
+        self._partitioning_scheme = scheme
+
+    @property
+    def file_system_type(self) -> Str:
+        """Type of a file system used on the partitions.
```

Comment:
I wonder if it would make sense to have these specified by an enum/constnats or some other way where one could easily check what are the valid values ?

Reply:
Well, Blivet will fail at some point if you specify an invalid value. We cannot validate these values in the DBus structure, because that would require to import objects from Blivet.